### PR TITLE
Add Lua 5.4 syntax highlighting and indentation tokenization

### PR DESCRIPTION
**Pull request by SentientCoffee**

### CODE REVIEW

File: src/langs/lua.jai [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+    identifier.data = t;
+
+    while t < max_t && is_alnum(t.*) {
+        t += 1;
+    }
+    if t >= max_t then t = max_t;
+    identifier.count = t - identifier.data;
+
+    return identifier;
+}
```
Comment (focus-editor) [inline src/langs/lua.jai:414]: Can the generic function be used here instead? If the only reason for having it is the custom tokenizer, we could make it inherit from `Tokenizer` with `#using #as` and then use the generic function (here and in other places)

File: src/langs/lua.jai [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+
+eat_white_space :: (using tokenizer: *Lua_Tokenizer) {
+    while t < max_t && is_white_space(t.*) {
+        t += 1;
+    }
+}
+
+is_hex :: inline (c: u8) -> bool {
+    return is_digit(c) || (c >= #char "a" && c <= #char "f") || (c >= #char "A" && c <= #char "F");
+}
```
Comment (focus-editor) [inline src/langs/lua.jai:430]: We already have a function for that don't we?

File: src/langs/lua.jai [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+
+is_white_space :: inline (char : u8) -> bool #no_aoc {
```
Comment (focus-editor) [inline src/langs/lua.jai:432]: Same question here. If this one is different for some reason, then please add a comment explaining why we had to copy it.

**Review by focus-editor — CHANGES_REQUESTED** [review]

**Comment by SentientCoffee** [comment]
The problem with the review was just that I copy pasted the file to keep it all contained in one file, I can delete the unneeded procs and resubmit.

File: src/langs/lua.jai (hunk 2 above)
Comment (SentientCoffee) [reply src/langs/lua.jai]: See below comment.

File: src/langs/lua.jai (hunk 3 above)
Comment (SentientCoffee) [reply src/langs/lua.jai]: See below comment.

File: src/langs/lua.jai (hunk 1 above)
Comment (SentientCoffee) [reply src/langs/lua.jai]: I've changed the `Lua_Tokenizer` to have `using #as base: Tokenizer`.

**Comment by focus-editor** [comment]
Thanks @SentientCoffee !

**Pull request merged**